src/tests/Common/XUnitLogChecker/XUnitLogChecker.cs

```diff
@@ -379,27 +459,46 @@ static TagResult[] GetOrderedTagMatches(Match[] openingTags, Match[] closingTags
         return result;
     }
 
-    static void PrintStackTracesFromDumps(string dumpsPath, string tempLogPath)
+    static void PrintStackTracesFromDumps(string testLogPath)
     {
         WriteLineTimestamp("Checking for dumps...");
+        IEnumerable<string>? dumpsFound = null;
 
-        // Read our newly fixed log to retrieve the time and date when the
-        // test was run. This is to exclude potentially existing older dumps
-        // that are not related to this test run.
-        XElement fixedLogTree = XElement.Load(tempLogPath);
-
-        // We know from the XUnitWrapperGenerator that the top element
-        // is the 'assembly' tag we're looking for.
-        var testRunDateTime = DateTime.ParseExact
-        (
-            fixedLogTree.Attribute("run-date-time").Value,
-            "yyyy-MM-dd HH:mm:ss",
-            System.Globalization.CultureInfo.InvariantCulture
-        );
+        // In CoreCLR, the test results log has the test's run time recorded.
+        // We extract it and use it to only retrieve the dumps that were created
+        // after that point in time. Any other dump present there is certainly
+        // unrelated to the currently being analyzed test.
+        //
+        // In Libraries, since we can't get said timestamp, we just retrieve
+        // all of the dumps.
 
-        IEnumerable<string> dumpsFound =
-            Directory.GetFiles(dumpsPath, "*coredump*.dmp")
-                     .Where(dmp => DateTime.Compare(File.GetCreationTime(dmp), testRunDateTime) >= 0);
+        if (!string.IsNullOrEmpty(testLogPath))
+        {
+            // Read our newly fixed log to retrieve the time and date when the
+            // test was run. This is to exclude potentially existing older dumps
+            // that are not related to this test run.
+            XElement fixedLogTree = XElement.Load(testLogPath);
+
+            // We know from the XUnitWrapperGenerator that the top element
+            // is the 'assembly' tag we're looking for.
+            var testRunDateTime = DateTime.ParseExact
+            (
+                fixedLogTree.Attribute("run-date-time").Value,
+                "yyyy-MM-dd HH:mm:ss",
+                System.Globalization.CultureInfo.InvariantCulture
+            );
+
+            dumpsFound = Directory
```

Comment:
Should we report and / or delete the obsolete dumps or do we know that the OS or Helix infra deals with that somehow?